rust/kcl-lib/src/execution/exec_ast.rs

```diff
@@ -1118,10 +1122,35 @@ impl Node<CallExpressionKw> {
                     None
                 };
 
+                let formals = func.args(false);
+                #[allow(clippy::iter_over_hash_type)]
```

Comment:
We shouldn't allow this because it will make the error non-deterministic, meaning that tests would be flaky. The simplest way is to change the type to `IndexSet`. This is the main thing I think we should change before merging.

Reply:
I was hoping we didn't rely on the ordering of errors, but I can just change to IndexMap

Reply:
Do we test multiple non-fatal errors on the Rust side? I suspect we don't, and that's why it's not currently an issue. But we should.